src/main/java/namor/template/calculator/RPNcalculator.java

```diff
@@ -0,0 +1,83 @@
+package namor.template.calculator;
+
+import namor.template.stack.ArrayStack;
+
+public class RPNcalculator {
+    private ArrayStack numbers = new ArrayStack();
+    Sign sign;
+
+    public int calculate(String rpnString){
+        int count = 0;
+        String[] retval = rpnString.split(" ");
+
+         while (retval.length > count){
+            if(retval[count].matches("^[0-9]+$")){
+                numbers.push(Integer.parseInt(retval[count]));
+            } else {
+                switch (sign.getValue(retval[count])){
+                    case ADDITION:
+                        numbers.push(addition());
+                        break;
+                    case SUBTRACTION:
+                        numbers.push(subtraction());
+                        break;
+                    case MULTIPLICATION:
+                        numbers.push(multiplication());
+                        break;
+                    case DIVISION:
+                        numbers.push(division());
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Input elements are wrong!");
+                }
+            }
+            count++;
+        }
+
+        return numbers.pop();
+    }
+
+    private int addition(){
+        int result;
+        int firstNumber;
+        int secondNumber;
+
+        secondNumber = numbers.pop();
+        firstNumber = numbers.pop();
+        result = firstNumber + secondNumber;
+        return result;
+    }
+
+    private int subtraction(){
+        int result;
+        int firstNumber;
+        int secondNumber;
```

Comment:
You can assign this variables from the beginning why o you need two aditional rows for it ? and also you can make them final. So to be sure they will not be changed